designs/debug_mode_multiple_exposed_ports.md

```diff
@@ -0,0 +1,122 @@
+Title: Template for design documents
+====================================
+
+What is the problem?
+--------------------
+Currently, there is one port is exposed from Docker instance when running lambda in debug mode. 
+This port is used to connect a debugger. In my case, I need two ports to be exposed due to Debugger 
+implementation specific (the Debugger connect to two sockets to collect different information). 
+
+What will be changed?
+---------------------
+SAM CLI has a ``--debug-port`` parameter that provide a port. This parameter is stored in DebugContext object.
+``DebugContext`` should store an array of ports instead of one single port. This array should be transformed
+into a map containing each stored port when passing to docker container arguments.
+
+Success criteria for the change
+-------------------------------
+All ports (comma separated) specified in ``--debug-port`` SAM CLI parameter should be exposed by docker container.
```

Comment:
Why comma separated? This seems like a good use-case for multiple options: `--debug-port 5600 --debug-port 5601`.

Reply:
Good point. Updated with `multiple=true` option parameter.